contracts/Identity.sol

```diff
@@ -1,15 +1,294 @@
-pragma solidity ^0.6.2;
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.6.9;
 
-import "./ERC734.sol";
-import "./IIdentity.sol";
+import "./interface/IIdentity.sol";
+import "./version/Version.sol";
+import "./storage/Storage.sol";
 
 /**
  * @dev Implementation of the `IERC734` "KeyHolder" and the `IERC735` "ClaimHolder" interfaces into a common Identity Contract.
+ * This implementation has a separate contract were it declares all storage, allowing for it to be used as an upgradable logic contract.
  */
-contract Identity is ERC734, IIdentity {
+contract Identity is Storage, IIdentity, Version {
+    bool private initialized = false;
+    bool private canInteract = true;
 
-    mapping (bytes32 => Claim) private claims;
-    mapping (uint256 => bytes32[]) private claimsByTopic;
+    constructor(address initialManagementKey, bool _isLibrary) public {
+        canInteract = !_isLibrary;
+
+        if (canInteract) {
+            __Identity_init(initialManagementKey);
+        } else {
+            initialized = true;
+        }
+    }
+
+    modifier delegatedOnly() {
+        require(canInteract == true, "Interacting with the library contract is forbidden.");
+        _;
+    }
+
+    function initialize(address initialManagementKey) public {
+        __Identity_init(initialManagementKey);
+    }
+
+    function _isConstructor() private view returns (bool) {
+        address self = address(this);
+        uint256 cs;
+        // solhint-disable-next-line no-inline-assembly
+        assembly { cs := extcodesize(self) }
+        return cs == 0;
+    }
+
+    event ExecutionFailed(uint256 indexed executionId, address indexed to, uint256 indexed value, bytes data);
```

Comment:
most of the time i prefer to put event declaration on the interfaces, in any case we should put all the events declarations at the same place